## 7.17 
模拟一个 $\phi=0.7$，$\theta=0.4$，$n=72$ 的 ARMA(1,1) 序列。

In [1]:
# 设置参数
phi = 0.7
theta = 0.4
n = 72

# 生成序列
set.seed(114514)
series=arima.sim(n=n,list(ar=phi,ma=-theta))

### (a) 
求 $\phi$ 和 $\theta$ 的矩估计。  

$\text{Solution.}$

对于 ARMA(1,1) 模型，其矩估计是通过样本自相关函数来估计模型参数,我们利用样本自相关系数与理论自相关系数之间的关系，解方程得到参数估计值。

对于 ARMA(1,1)，有： 

$$\hat{\phi}=r_2/r_1 \quad (1)$$
$$r_1 = \frac{(1-\hat{\theta}\hat{{\phi}})(\hat{\phi}-\hat{\theta})}{1-2\hat{\theta}\hat{\phi}+\hat{\theta}^2} \quad (2)$$

这里需要先后解这两个方程来估计 $\hat{\phi}$ 和 $\hat{\theta}$ :

In [2]:
acf(series, lag.max = 5, plot = FALSE)


Autocorrelations of series 'series', by lag

    0     1     2     3     4     5 
1.000 0.287 0.220 0.213 0.093 0.263 

In [3]:
r1 = acf(series, lag.max = 5, plot = FALSE)$acf[[2]]
r2 = acf(series, lag.max = 5, plot = FALSE)$acf[[3]]

In [4]:
r1

[1] 0.2872912

基于$r_1, r_2$，我们可以解方程得到 $\hat{\phi}$ 和 $\hat{\theta}$ 的估计值。

In [5]:
hat_phi = r2/r1

In [6]:
hat_phi

[1] 0.7672112

$\hat{\phi}$ 如上展示，  

带入$r_1$ 的公式中，解方程得到 $\hat{\theta}$ 的估计值为复数解，因此该模型不适合使用矩估计对参数$\theta$作估计。

### (b) 
求 $\phi$ 和 $\theta$ 的条件最小二乘估计，并与 (a) 比较。  

$\text{Solution.}$

In [7]:
arima(series,order=c(1,0,1),method='CSS')


Call:
arima(x = series, order = c(1, 0, 1), method = "CSS")

Coefficients:
         ar1      ma1  intercept
      0.7226  -0.4719     0.1295
s.e.  0.2070   0.2493     0.2376

sigma^2 estimated as 1.057:  part log likelihood = -104.15

可知，$\hat{\phi}_{LS }= 0.7226\space , \space \hat{\theta}_{LS} = 0.4719$  

$\hat{\phi}_{LS}$与（a）中差别不算太大。

### (c) 
求 $\phi$ 和 $\theta$ 的极大似然估计，并与 (a) 和 (b) 比较。

$\text{Solution.}$

In [8]:
arima(series,order=c(1,0,1),method='ML')


Call:
arima(x = series, order = c(1, 0, 1), method = "ML")

Coefficients:
         ar1      ma1  intercept
      0.8262  -0.6124     0.0511
s.e.  0.1588   0.2092     0.2611

sigma^2 estimated as 1.023:  log likelihood = -103.09,  aic = 214.18

可知，$\hat{\phi}_{ML}= 0.8262\space , \space \hat{\theta}_{ML} = 0.6124$

相较而言，这次模拟的偏误算是较大了

### (d) 
在选取相同参数和样本规模的情况下，使用新的模拟序列重复 (a)、(b) 和 (c)，并将本次与前次的模拟结果进行比较。  

$\text{Solution.}$

In [24]:
# 生成序列
set.seed(99993)
series=arima.sim(n=n,list(ar=phi,ma=-theta))

In [25]:
acf(series, lag.max = 5, plot = FALSE)


Autocorrelations of series 'series', by lag

     0      1      2      3      4      5 
 1.000  0.355  0.240  0.084  0.130 -0.044 

In [26]:
r1 = acf(series, lag.max = 5, plot = FALSE)$acf[[2]]
r2 = acf(series, lag.max = 5, plot = FALSE)$acf[[3]]

In [27]:
hat_phi = r2/r1

In [29]:
# 矩估计
hat_phi

[1] 0.6774238

In [31]:
# 最小二乘
arima(series,order=c(1,0,1),method='CSS')


Call:
arima(x = series, order = c(1, 0, 1), method = "CSS")

Coefficients:
         ar1     ma1  intercept
      0.3432  0.0202     0.2393
s.e.  0.2831  0.2685     0.2052

sigma^2 estimated as 1.252:  part log likelihood = -110.25

In [33]:
# 极大似然估计
arima(series,order=c(1,0,1),method='ML')


Call:
arima(x = series, order = c(1, 0, 1), method = "ML")

Coefficients:
         ar1      ma1  intercept
      0.5938  -0.2655     0.1959
s.e.  0.2292   0.2736     0.2359

sigma^2 estimated as 1.257:  log likelihood = -110.49,  aic = 228.97

与前一次模拟相比，这一次通过矩估计得到的$\hat{\phi}$与理论值相差更小，总的来说两次模拟都还算准确。  

然而，通过最小二乘和极大似然估计得到的$\hat{\phi}$与理论值并不那么接近，这有点奇怪。  

原因可能是多方面的。首先，ARMA(1,1)模型是一个简单的模型，其参数估计可能受到样本规模和噪声的影响。其次，ARMA(1,1)模型的参数估计可能受到模型假设的限制，例如，模型的阶数和参数的约束条件。最后，ARMA(1,1)模型的参数估计可能受到数据集的特性和分布的影响。  